server/repository/src/db_diesel/assets/asset.rs

```diff
@@ -207,42 +183,6 @@ fn create_filtered_query(filter: Option<AssetFilter>) -> BoxedAssetQuery {
         apply_equal_filter!(query, catalogue_item_id, asset_dsl::asset_catalogue_item_id);
         apply_date_filter!(query, installation_date, asset_dsl::installation_date);
         apply_date_filter!(query, replacement_date, asset_dsl::replacement_date);
-
-        if let Some(category_id) = category_id {
-            let mut sub_query = asset_catalogue_item_dsl::asset_catalogue_item
-                .select(asset_catalogue_item_dsl::id.nullable())
-                .into_boxed();
-            apply_equal_filter!(
-                sub_query,
-                Some(category_id),
-                asset_catalogue_item_dsl::asset_category_id
-            );
-            query = query.filter(asset_dsl::asset_catalogue_item_id.eq_any(sub_query));
-        }
-
-        if let Some(class_id) = class_id {
-            let mut sub_query = asset_catalogue_item_dsl::asset_catalogue_item
-                .select(asset_catalogue_item_dsl::id.nullable())
-                .into_boxed();
-            apply_equal_filter!(
-                sub_query,
-                Some(class_id),
-                asset_catalogue_item_dsl::asset_class_id
-            );
-            query = query.filter(asset_dsl::asset_catalogue_item_id.eq_any(sub_query));
-        }
-
-        if let Some(type_id) = type_id {
-            let mut sub_query = asset_catalogue_item_dsl::asset_catalogue_item
-                .select(asset_catalogue_item_dsl::id.nullable())
-                .into_boxed();
-            apply_equal_filter!(
-                sub_query,
-                Some(type_id),
-                asset_catalogue_item_dsl::asset_type_id
-            );
-            query = query.filter(asset_dsl::asset_catalogue_item_id.eq_any(sub_query));
-        }
```

Comment:
These filters are moved into asset_catalogue_item because that table now contains the asset_type_id, asset_class_id, and asset_catagory_id.

All these test changes were made in the last commit. I've added them to this PR so that the tests would be fixed.